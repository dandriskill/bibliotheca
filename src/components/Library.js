import React from 'react';
import '../styles/Library.css';

import Book from './Book';

function Library({
  books,
  openModal,
  closeModal,
  updateBook,
  deleteBook,
}) {
  return (
    <div className='Library'>
      {books ? (
        <ul className='books'>
          {books.map((book) => {
            return (
              <Book
                key={book.id}
                book={book}
                openModal={openModal}
                closeModal={closeModal}
                updateBook={updateBook}
                deleteBook={deleteBook}
              />
            );
          })}
        </ul>
      ) : (
        <div className="fallback-text">No volumes available.</div>
      )}
    </div>
  );
}

export default Library;