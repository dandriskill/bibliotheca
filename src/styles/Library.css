.Library {
  margin: 0 50px;
  height: 50%;
  overflow-y: scroll;
  overflow-x: hidden;
}

.books {
  padding: 0;
  margin: 0;
}

.fallback-text {
  margin: 30px 0;
}