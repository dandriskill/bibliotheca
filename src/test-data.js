export const testBooks = [
  {
    id: 1,
    title: 'Republic',
    author: 'Plato',
    pages: '334',
    available: 'Yes',
    overdue: 'No',
  },
  {
    id: 2,
    title: 'Oddyssey',
    author: 'Homer',
    pages: '384',
    available: 'No',
    overdue: 'Yes',
  },
  {
    id: 3,
    title: 'The Prince',
    author: 'Niccol√≤ Machiavelli',
    pages: '71',
    available: 'No',
    overdue: 'No',
  },
];