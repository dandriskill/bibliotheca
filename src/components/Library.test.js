import React from 'react';
import { render } from '@testing-library/react';
import Library from './Library';
import { testBooks } from '../test-data';

describe('<Library />', () => {
  test('matches snapshot', () => {
    const container = render(<Library books={testBooks} />);
    expect(container).toMatchSnapshot();
  });

  test('shows books when "books" prop is populated', () => {
    const { queryByText } = render(<Library books={testBooks} />);
    expect(queryByText('The Prince')).toBeTruthy();
    expect(queryByText('Republic')).toBeTruthy();
    expect(queryByText('Oddyssey')).toBeTruthy();
    expect(queryByText('No volumes available.')).toBeFalsy();
  });

  test('shows fallback text when "books" prop is not populated', () => {
    const { queryByText } = render(<Library books={[]} />);
    expect(queryByText('Prince')).toBeFalsy();
    expect(queryByText('Republic')).toBeFalsy();
    expect(queryByText('Oddyssey')).toBeFalsy();
    expect(queryByText('No volumes available.')).toBeTruthy();
  });
});