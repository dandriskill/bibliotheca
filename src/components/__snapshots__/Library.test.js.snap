// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Library /> matches snapshot 1`] = `
Object {
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <div
        class="Library"
      >
        <ul
          class="books"
        >
          <li
            class="Book"
          >
            <div
              class="book-title"
            >
              Republic
            </div>
            <div
              class="book-author"
            >
              Plato
            </div>
            <div
              class="book-pages"
            >
              334
            </div>
            <div
              class="book-available"
            >
              Yes
            </div>
            <div
              class="book-overdue"
            >
              Yes
            </div>
            <div
              class="book-tools"
            >
              <div>
                <div
                  title="Edit"
                >
                  <svg
                    class="edit-button"
                    fill="#282c34"
                    viewBox="0 0 512 512"
                  >
                    <path
                      d="M479.999 0C263.145 0 84.777 166.827 65.876 378.837c36.032-50.816 80.171-99.712 135.275-155.925 4.096-4.224 10.88-4.309 15.083-.149 4.203 4.117 4.267 10.88.149 15.083-8.299 8.448-16.725 17.216-25.195 26.091-4.843 5.056-9.579 10.091-14.229 15.04-.853.917-1.707 1.835-2.56 2.773C106.303 354.667 57.62 418.133 22.27 496.96c-2.411 5.376 0 11.691 5.355 14.101 1.429.64 2.901.939 4.373.939 4.075 0 7.979-2.347 9.749-6.293 12.587-28.053 27.2-54.08 43.627-79.403 148.992-3.819 284.608-86.699 355.691-218.581a10.726 10.726 0 00-.235-10.539 10.751 10.751 0 00-9.173-5.184h-39.509l70.848-40.491a10.686 10.686 0 004.8-5.781c15.189-44.16 22.869-89.6 22.869-135.061C490.665 4.779 485.887 0 479.999 0z"
                    />
                  </svg>
                </div>
              </div>
              <div>
                <div
                  title="Delete"
                >
                  <svg
                    class="delete-button"
                    fill="#282c34"
                    viewBox="0 0 512 512"
                  >
                    <path
                      d="M446.906 299.769c-5.865-76.359-41.417-124.21-72.781-166.436C345.083 94.241 320 60.483 320 10.685c0-4-2.24-7.656-5.792-9.489-3.563-1.844-7.844-1.542-11.083.812-47.104 33.706-86.406 90.515-100.135 144.719-9.531 37.737-10.792 80.161-10.969 108.18-43.5-9.291-53.354-74.359-53.458-75.068a10.72 10.72 0 00-5.552-7.916c-3.031-1.583-6.594-1.698-9.667-.177-2.281 1.104-55.99 28.394-59.115 137.355-.219 3.625-.229 7.261-.229 10.896 0 105.857 86.135 191.987 192 191.987.146.01.302.031.427 0h.135C362.167 511.681 448 425.667 448 319.997c0-5.323-1.094-20.228-1.094-20.228zM256 490.652c-35.292 0-64-30.581-64-68.172 0-1.281-.01-2.573.083-4.156.427-15.853 3.438-26.675 6.74-33.873 6.188 13.291 17.25 25.509 35.219 25.509a10.66 10.66 0 0010.667-10.666c0-15.186.313-32.706 4.094-48.518 3.365-14.02 11.406-28.936 21.594-40.893 4.531 15.52 13.365 28.081 21.99 40.341 12.344 17.54 25.104 35.675 27.344 66.6.135 1.833.271 3.677.271 5.656-.002 37.59-28.71 68.172-64.002 68.172z"
                    />
                  </svg>
                </div>
              </div>
            </div>
          </li>
          <li
            class="Book"
          >
            <div
              class="book-title"
            >
              Oddyssey
            </div>
            <div
              class="book-author"
            >
              Homer
            </div>
            <div
              class="book-pages"
            >
              384
            </div>
            <div
              class="book-available"
            >
              Yes
            </div>
            <div
              class="book-overdue"
            >
              Yes
            </div>
            <div
              class="book-tools"
            >
              <div>
                <div
                  title="Edit"
                >
                  <svg
                    class="edit-button"
                    fill="#282c34"
                    viewBox="0 0 512 512"
                  >
                    <path
                      d="M479.999 0C263.145 0 84.777 166.827 65.876 378.837c36.032-50.816 80.171-99.712 135.275-155.925 4.096-4.224 10.88-4.309 15.083-.149 4.203 4.117 4.267 10.88.149 15.083-8.299 8.448-16.725 17.216-25.195 26.091-4.843 5.056-9.579 10.091-14.229 15.04-.853.917-1.707 1.835-2.56 2.773C106.303 354.667 57.62 418.133 22.27 496.96c-2.411 5.376 0 11.691 5.355 14.101 1.429.64 2.901.939 4.373.939 4.075 0 7.979-2.347 9.749-6.293 12.587-28.053 27.2-54.08 43.627-79.403 148.992-3.819 284.608-86.699 355.691-218.581a10.726 10.726 0 00-.235-10.539 10.751 10.751 0 00-9.173-5.184h-39.509l70.848-40.491a10.686 10.686 0 004.8-5.781c15.189-44.16 22.869-89.6 22.869-135.061C490.665 4.779 485.887 0 479.999 0z"
                    />
                  </svg>
                </div>
              </div>
              <div>
                <div
                  title="Delete"
                >
                  <svg
                    class="delete-button"
                    fill="#282c34"
                    viewBox="0 0 512 512"
                  >
                    <path
                      d="M446.906 299.769c-5.865-76.359-41.417-124.21-72.781-166.436C345.083 94.241 320 60.483 320 10.685c0-4-2.24-7.656-5.792-9.489-3.563-1.844-7.844-1.542-11.083.812-47.104 33.706-86.406 90.515-100.135 144.719-9.531 37.737-10.792 80.161-10.969 108.18-43.5-9.291-53.354-74.359-53.458-75.068a10.72 10.72 0 00-5.552-7.916c-3.031-1.583-6.594-1.698-9.667-.177-2.281 1.104-55.99 28.394-59.115 137.355-.219 3.625-.229 7.261-.229 10.896 0 105.857 86.135 191.987 192 191.987.146.01.302.031.427 0h.135C362.167 511.681 448 425.667 448 319.997c0-5.323-1.094-20.228-1.094-20.228zM256 490.652c-35.292 0-64-30.581-64-68.172 0-1.281-.01-2.573.083-4.156.427-15.853 3.438-26.675 6.74-33.873 6.188 13.291 17.25 25.509 35.219 25.509a10.66 10.66 0 0010.667-10.666c0-15.186.313-32.706 4.094-48.518 3.365-14.02 11.406-28.936 21.594-40.893 4.531 15.52 13.365 28.081 21.99 40.341 12.344 17.54 25.104 35.675 27.344 66.6.135 1.833.271 3.677.271 5.656-.002 37.59-28.71 68.172-64.002 68.172z"
                    />
                  </svg>
                </div>
              </div>
            </div>
          </li>
          <li
            class="Book"
          >
            <div
              class="book-title"
            >
              The Prince
            </div>
            <div
              class="book-author"
            >
              Niccol√≤ Machiavelli
            </div>
            <div
              class="book-pages"
            >
              71
            </div>
            <div
              class="book-available"
            >
              Yes
            </div>
            <div
              class="book-overdue"
            >
              Yes
            </div>
            <div
              class="book-tools"
            >
              <div>
                <div
                  title="Edit"
                >
                  <svg
                    class="edit-button"
                    fill="#282c34"
                    viewBox="0 0 512 512"
                  >
                    <path
                      d="M479.999 0C263.145 0 84.777 166.827 65.876 378.837c36.032-50.816 80.171-99.712 135.275-155.925 4.096-4.224 10.88-4.309 15.083-.149 4.203 4.117 4.267 10.88.149 15.083-8.299 8.448-16.725 17.216-25.195 26.091-4.843 5.056-9.579 10.091-14.229 15.04-.853.917-1.707 1.835-2.56 2.773C106.303 354.667 57.62 418.133 22.27 496.96c-2.411 5.376 0 11.691 5.355 14.101 1.429.64 2.901.939 4.373.939 4.075 0 7.979-2.347 9.749-6.293 12.587-28.053 27.2-54.08 43.627-79.403 148.992-3.819 284.608-86.699 355.691-218.581a10.726 10.726 0 00-.235-10.539 10.751 10.751 0 00-9.173-5.184h-39.509l70.848-40.491a10.686 10.686 0 004.8-5.781c15.189-44.16 22.869-89.6 22.869-135.061C490.665 4.779 485.887 0 479.999 0z"
                    />
                  </svg>
                </div>
              </div>
              <div>
                <div
                  title="Delete"
                >
                  <svg
                    class="delete-button"
                    fill="#282c34"
                    viewBox="0 0 512 512"
                  >
                    <path
                      d="M446.906 299.769c-5.865-76.359-41.417-124.21-72.781-166.436C345.083 94.241 320 60.483 320 10.685c0-4-2.24-7.656-5.792-9.489-3.563-1.844-7.844-1.542-11.083.812-47.104 33.706-86.406 90.515-100.135 144.719-9.531 37.737-10.792 80.161-10.969 108.18-43.5-9.291-53.354-74.359-53.458-75.068a10.72 10.72 0 00-5.552-7.916c-3.031-1.583-6.594-1.698-9.667-.177-2.281 1.104-55.99 28.394-59.115 137.355-.219 3.625-.229 7.261-.229 10.896 0 105.857 86.135 191.987 192 191.987.146.01.302.031.427 0h.135C362.167 511.681 448 425.667 448 319.997c0-5.323-1.094-20.228-1.094-20.228zM256 490.652c-35.292 0-64-30.581-64-68.172 0-1.281-.01-2.573.083-4.156.427-15.853 3.438-26.675 6.74-33.873 6.188 13.291 17.25 25.509 35.219 25.509a10.66 10.66 0 0010.667-10.666c0-15.186.313-32.706 4.094-48.518 3.365-14.02 11.406-28.936 21.594-40.893 4.531 15.52 13.365 28.081 21.99 40.341 12.344 17.54 25.104 35.675 27.344 66.6.135 1.833.271 3.677.271 5.656-.002 37.59-28.71 68.172-64.002 68.172z"
                    />
                  </svg>
                </div>
              </div>
            </div>
          </li>
        </ul>
      </div>
    </div>
  </body>,
  "container": <div>
    <div
      class="Library"
    >
      <ul
        class="books"
      >
        <li
          class="Book"
        >
          <div
            class="book-title"
          >
            Republic
          </div>
          <div
            class="book-author"
          >
            Plato
          </div>
          <div
            class="book-pages"
          >
            334
          </div>
          <div
            class="book-available"
          >
            Yes
          </div>
          <div
            class="book-overdue"
          >
            Yes
          </div>
          <div
            class="book-tools"
          >
            <div>
              <div
                title="Edit"
              >
                <svg
                  class="edit-button"
                  fill="#282c34"
                  viewBox="0 0 512 512"
                >
                  <path
                    d="M479.999 0C263.145 0 84.777 166.827 65.876 378.837c36.032-50.816 80.171-99.712 135.275-155.925 4.096-4.224 10.88-4.309 15.083-.149 4.203 4.117 4.267 10.88.149 15.083-8.299 8.448-16.725 17.216-25.195 26.091-4.843 5.056-9.579 10.091-14.229 15.04-.853.917-1.707 1.835-2.56 2.773C106.303 354.667 57.62 418.133 22.27 496.96c-2.411 5.376 0 11.691 5.355 14.101 1.429.64 2.901.939 4.373.939 4.075 0 7.979-2.347 9.749-6.293 12.587-28.053 27.2-54.08 43.627-79.403 148.992-3.819 284.608-86.699 355.691-218.581a10.726 10.726 0 00-.235-10.539 10.751 10.751 0 00-9.173-5.184h-39.509l70.848-40.491a10.686 10.686 0 004.8-5.781c15.189-44.16 22.869-89.6 22.869-135.061C490.665 4.779 485.887 0 479.999 0z"
                  />
                </svg>
              </div>
            </div>
            <div>
              <div
                title="Delete"
              >
                <svg
                  class="delete-button"
                  fill="#282c34"
                  viewBox="0 0 512 512"
                >
                  <path
                    d="M446.906 299.769c-5.865-76.359-41.417-124.21-72.781-166.436C345.083 94.241 320 60.483 320 10.685c0-4-2.24-7.656-5.792-9.489-3.563-1.844-7.844-1.542-11.083.812-47.104 33.706-86.406 90.515-100.135 144.719-9.531 37.737-10.792 80.161-10.969 108.18-43.5-9.291-53.354-74.359-53.458-75.068a10.72 10.72 0 00-5.552-7.916c-3.031-1.583-6.594-1.698-9.667-.177-2.281 1.104-55.99 28.394-59.115 137.355-.219 3.625-.229 7.261-.229 10.896 0 105.857 86.135 191.987 192 191.987.146.01.302.031.427 0h.135C362.167 511.681 448 425.667 448 319.997c0-5.323-1.094-20.228-1.094-20.228zM256 490.652c-35.292 0-64-30.581-64-68.172 0-1.281-.01-2.573.083-4.156.427-15.853 3.438-26.675 6.74-33.873 6.188 13.291 17.25 25.509 35.219 25.509a10.66 10.66 0 0010.667-10.666c0-15.186.313-32.706 4.094-48.518 3.365-14.02 11.406-28.936 21.594-40.893 4.531 15.52 13.365 28.081 21.99 40.341 12.344 17.54 25.104 35.675 27.344 66.6.135 1.833.271 3.677.271 5.656-.002 37.59-28.71 68.172-64.002 68.172z"
                  />
                </svg>
              </div>
            </div>
          </div>
        </li>
        <li
          class="Book"
        >
          <div
            class="book-title"
          >
            Oddyssey
          </div>
          <div
            class="book-author"
          >
            Homer
          </div>
          <div
            class="book-pages"
          >
            384
          </div>
          <div
            class="book-available"
          >
            Yes
          </div>
          <div
            class="book-overdue"
          >
            Yes
          </div>
          <div
            class="book-tools"
          >
            <div>
              <div
                title="Edit"
              >
                <svg
                  class="edit-button"
                  fill="#282c34"
                  viewBox="0 0 512 512"
                >
                  <path
                    d="M479.999 0C263.145 0 84.777 166.827 65.876 378.837c36.032-50.816 80.171-99.712 135.275-155.925 4.096-4.224 10.88-4.309 15.083-.149 4.203 4.117 4.267 10.88.149 15.083-8.299 8.448-16.725 17.216-25.195 26.091-4.843 5.056-9.579 10.091-14.229 15.04-.853.917-1.707 1.835-2.56 2.773C106.303 354.667 57.62 418.133 22.27 496.96c-2.411 5.376 0 11.691 5.355 14.101 1.429.64 2.901.939 4.373.939 4.075 0 7.979-2.347 9.749-6.293 12.587-28.053 27.2-54.08 43.627-79.403 148.992-3.819 284.608-86.699 355.691-218.581a10.726 10.726 0 00-.235-10.539 10.751 10.751 0 00-9.173-5.184h-39.509l70.848-40.491a10.686 10.686 0 004.8-5.781c15.189-44.16 22.869-89.6 22.869-135.061C490.665 4.779 485.887 0 479.999 0z"
                  />
                </svg>
              </div>
            </div>
            <div>
              <div
                title="Delete"
              >
                <svg
                  class="delete-button"
                  fill="#282c34"
                  viewBox="0 0 512 512"
                >
                  <path
                    d="M446.906 299.769c-5.865-76.359-41.417-124.21-72.781-166.436C345.083 94.241 320 60.483 320 10.685c0-4-2.24-7.656-5.792-9.489-3.563-1.844-7.844-1.542-11.083.812-47.104 33.706-86.406 90.515-100.135 144.719-9.531 37.737-10.792 80.161-10.969 108.18-43.5-9.291-53.354-74.359-53.458-75.068a10.72 10.72 0 00-5.552-7.916c-3.031-1.583-6.594-1.698-9.667-.177-2.281 1.104-55.99 28.394-59.115 137.355-.219 3.625-.229 7.261-.229 10.896 0 105.857 86.135 191.987 192 191.987.146.01.302.031.427 0h.135C362.167 511.681 448 425.667 448 319.997c0-5.323-1.094-20.228-1.094-20.228zM256 490.652c-35.292 0-64-30.581-64-68.172 0-1.281-.01-2.573.083-4.156.427-15.853 3.438-26.675 6.74-33.873 6.188 13.291 17.25 25.509 35.219 25.509a10.66 10.66 0 0010.667-10.666c0-15.186.313-32.706 4.094-48.518 3.365-14.02 11.406-28.936 21.594-40.893 4.531 15.52 13.365 28.081 21.99 40.341 12.344 17.54 25.104 35.675 27.344 66.6.135 1.833.271 3.677.271 5.656-.002 37.59-28.71 68.172-64.002 68.172z"
                  />
                </svg>
              </div>
            </div>
          </div>
        </li>
        <li
          class="Book"
        >
          <div
            class="book-title"
          >
            The Prince
          </div>
          <div
            class="book-author"
          >
            Niccol√≤ Machiavelli
          </div>
          <div
            class="book-pages"
          >
            71
          </div>
          <div
            class="book-available"
          >
            Yes
          </div>
          <div
            class="book-overdue"
          >
            Yes
          </div>
          <div
            class="book-tools"
          >
            <div>
              <div
                title="Edit"
              >
                <svg
                  class="edit-button"
                  fill="#282c34"
                  viewBox="0 0 512 512"
                >
                  <path
                    d="M479.999 0C263.145 0 84.777 166.827 65.876 378.837c36.032-50.816 80.171-99.712 135.275-155.925 4.096-4.224 10.88-4.309 15.083-.149 4.203 4.117 4.267 10.88.149 15.083-8.299 8.448-16.725 17.216-25.195 26.091-4.843 5.056-9.579 10.091-14.229 15.04-.853.917-1.707 1.835-2.56 2.773C106.303 354.667 57.62 418.133 22.27 496.96c-2.411 5.376 0 11.691 5.355 14.101 1.429.64 2.901.939 4.373.939 4.075 0 7.979-2.347 9.749-6.293 12.587-28.053 27.2-54.08 43.627-79.403 148.992-3.819 284.608-86.699 355.691-218.581a10.726 10.726 0 00-.235-10.539 10.751 10.751 0 00-9.173-5.184h-39.509l70.848-40.491a10.686 10.686 0 004.8-5.781c15.189-44.16 22.869-89.6 22.869-135.061C490.665 4.779 485.887 0 479.999 0z"
                  />
                </svg>
              </div>
            </div>
            <div>
              <div
                title="Delete"
              >
                <svg
                  class="delete-button"
                  fill="#282c34"
                  viewBox="0 0 512 512"
                >
                  <path
                    d="M446.906 299.769c-5.865-76.359-41.417-124.21-72.781-166.436C345.083 94.241 320 60.483 320 10.685c0-4-2.24-7.656-5.792-9.489-3.563-1.844-7.844-1.542-11.083.812-47.104 33.706-86.406 90.515-100.135 144.719-9.531 37.737-10.792 80.161-10.969 108.18-43.5-9.291-53.354-74.359-53.458-75.068a10.72 10.72 0 00-5.552-7.916c-3.031-1.583-6.594-1.698-9.667-.177-2.281 1.104-55.99 28.394-59.115 137.355-.219 3.625-.229 7.261-.229 10.896 0 105.857 86.135 191.987 192 191.987.146.01.302.031.427 0h.135C362.167 511.681 448 425.667 448 319.997c0-5.323-1.094-20.228-1.094-20.228zM256 490.652c-35.292 0-64-30.581-64-68.172 0-1.281-.01-2.573.083-4.156.427-15.853 3.438-26.675 6.74-33.873 6.188 13.291 17.25 25.509 35.219 25.509a10.66 10.66 0 0010.667-10.666c0-15.186.313-32.706 4.094-48.518 3.365-14.02 11.406-28.936 21.594-40.893 4.531 15.52 13.365 28.081 21.99 40.341 12.344 17.54 25.104 35.675 27.344 66.6.135 1.833.271 3.677.271 5.656-.002 37.59-28.71 68.172-64.002 68.172z"
                  />
                </svg>
              </div>
            </div>
          </div>
        </li>
      </ul>
    </div>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;
